src/Histogram.hpp

```diff
@@ -59,29 +64,31 @@ class shasta::Histogram : public vector<size_t> {
 class shasta::Histogram2{
 public:
     /// Attributes ///
-    const double start;
-    const double stop;
-    const size_t binCount;
+    double start;
+    double stop;
+    size_t binCount;
     const double binSize;
-    vector<uint64_t> histogram;
+    deque<uint64_t> histogram;
     bool unboundedLeft;
     bool unboundedRight;
+    bool dynamicBounds;
 
     /// Methods ///
     Histogram2(
```

Comment:
Perhaps add a few lines a comment explaining the functionality of this class and how the various constructor parameters interact with each other?